hal/ticker_api.h

```diff
@@ -24,6 +24,21 @@
 #include <stdbool.h>
 #include "device.h"
 
+/**
+ * Number of microseconds in a second
+ */
+#define MICROSECONDS_IN_SECOND  (us_timestamp_t)1000000
+
+/**
+ * Converts seconds to microseconds
+ */
+#define SECONDS_TO_MICROSECONDS(SECONDS)   (us_timestamp_t)(MICROSECONDS_IN_SECOND * SECONDS)
+
+/**
+ * Converts microseconds to seconds
+ */
+#define MICROSECONDS_TO_SECONDS(MICROSECONDS) (s_timestamp_t)(MICROSECONDS / MICROSECONDS_IN_SECOND)
```

Comment:
Define the macros where they are needed. 
In addition, only SECONDS_TO_MICROSECONDS is used in actual code (not considering test code).

Reply:
It was defined where typedef related to timing are defined to be accessible along with all timing stuff.
Even if it is only used in test code SECONDS_TO_MICROSECONDS is a nice abstraction that applications can also use. The conversion from seconds to microseconds happened two places so using the macro makes sense. It will avoid users in the future doing a conversion and using `1000000f` again.

Reply:
If you feel strongly about the location of the macros, 49e1c52

Reply:
Great, thanks.